internal/rest/avi_obj_vs.go

```diff
@@ -847,8 +850,12 @@ func (rest *RestOperations) isHostPresentInSharedPool(hostname string, parentVs
 
 func (rest *RestOperations) GetIPAddrsFromCache(vsCache *avicache.AviVsCache) []string {
 	var IPAddrs []string
-	if len(vsCache.VSVipKeyCollection) == 0 {
-		parentVSKey := vsCache.ParentVSRef
+	vsCacheCopy, ok := vsCache.GetVSCopy()
```

Comment:
I did not get purpose behind this change. is new variable required over here?

Reply:
yes, since we are passing the pointer of child VS and overwriting the pointer with the parent VS object.